sonar-scm-git-plugin/src/test/java/org/sonarsource/scm/git/GitIgnoreCommandTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * SonarQube :: Plugins :: SCM :: Git
+ * Copyright (C) 2014-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.scm.git;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import org.eclipse.jgit.api.Git;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarsource.scm.git.Utils.javaUnzip;
+
+public class GitIgnoreCommandTest {
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void ignored_files_should_match_files_ignored_by_git() throws IOException {
+    Path projectDir = temp.newFolder().toPath();
+    javaUnzip(new File("test-repos/ignore-git.zip"), projectDir.toFile());
+
+    Path baseDir = projectDir.resolve("ignore-git");
+    GitIgnoreCommand underTest = new GitIgnoreCommand();
+    underTest.init(baseDir);
+
+    assertThat(underTest.isIgnored(baseDir.resolve(".gitignore"))).isFalse();
+    assertThat(underTest.isIgnored(baseDir.resolve("pom.xml"))).isFalse();
+    assertThat(underTest.isIgnored(baseDir.resolve("src/main/java/org/dummy/.gitignore"))).isFalse();
+    assertThat(underTest.isIgnored(baseDir.resolve("src/main/java/org/dummy/AnotherDummy.java"))).isFalse();
+    assertThat(underTest.isIgnored(baseDir.resolve("src/test/java/org/dummy/AnotherDummyTest.java"))).isFalse();
+
+    assertThat(underTest.isIgnored(baseDir.resolve("src/main/java/org/dummy/Dummy.java"))).isTrue();
+    assertThat(underTest.isIgnored(baseDir.resolve("target"))).isTrue();
+  }
+
+  @Test
+  public void only_files_from_baseDir_should_be_indexed_when_baseDir_is_subdirectory_of_git_dir() throws IOException {
```

Comment:
This is a duplicate of the `dont_index_files_outside_basedir()` test. It think it can be removed.